alibi_detect/utils/pytorch/kernels.py

```diff
@@ -68,16 +68,24 @@ def __init__(
     def sigma(self) -> torch.Tensor:
         return self.log_sigma.exp()
 
-    def forward(self, x: Union[np.ndarray, torch.Tensor], y: Union[np.ndarray, torch.Tensor],
-                infer_sigma: bool = False) -> torch.Tensor:
+    def forward(self, x: Union[np.ndarray, torch.Tensor],
+                y: Union[np.ndarray, torch.Tensor],
+                infer_sigma: bool = False,
+                pairwise: bool = True) -> torch.Tensor:
 
         x, y = torch.as_tensor(x), torch.as_tensor(y)
-        dist = distance.squared_pairwise_distance(x.flatten(1), y.flatten(1))  # [Nx, Ny]
+        if pairwise:
+            dist = distance.squared_pairwise_distance(x.flatten(1), y.flatten(1))  # [Nx, Ny]
+        else:
+            dist = distance.squared_distance(x.flatten(1), y.flatten(1))  # [N, 1]
 
         if infer_sigma or self.init_required:
             if self.trainable and infer_sigma:
                 raise ValueError("Gradients cannot be computed w.r.t. an inferred sigma value")
-            sigma = self.init_sigma_fn(x, y, dist)
+            if pairwise:
+                sigma = self.init_sigma_fn(x, y, dist)
+            else:
+                sigma = (.5 * dist.flatten().sort().values[dist.shape[0] // 2 - 1].unsqueeze(dim=-1)) ** .5
```

Comment:
Again I think we can avoid the hard-coding of this behaviour and fall back on `self.init_sigma_fn` but with the desired linear detector behaviour.

Reply:
Slightly tricky as the default `init_sigma_fn` is used by other detectors. Might be easier to keep the additional line here?